monai_ex/transforms/spatial/array.py

```diff
@@ -319,3 +323,71 @@ def __call__(self, img: NdarrayOrTensor, affine: Optional[np.ndarray] = None):
         if self.image_only:
             return output_data
         return output_data, affine, new_affine
+
+
+class RandomDrop(Randomizable, Transform):
+    """ Drop a number of patchs on the centerline of curvilinear structure randomly.
+
+    Args:
+        roi_number (int): Number of the drop roi.
+        roi_size (int): size of the drop ROI.
+        random_seed (int) : set the random state with an integer seed.
+    """
+    def __init__(
+        self,
+        roi_number: int,
+        roi_size: int,
+        random_state: int = 20221201
+    ) -> None:
+        self.roi_num = roi_number
+        self.roi_size = roi_size
+        self.random_seed = random_state
+
+    def __call__(
+        self,
+        img: NdarrayOrTensor,
+        cline: NdarrayOrTensor
+    ) -> NdarrayOrTensor:
+        """
+            Args:
+            img (NdarrayOrTensor): The image to drop patches.
+            cline (NdarrayOrTensor): The centerline of curvilinear structure, it can be extracted with `monai_ex.transforms.ExtractCenterline`
+        """
+        data = deepcopy(img)
+        dropped_data = []
+        c_points = np.nonzero(cline)
+        random.seed(self.random_seed)
+        roi_centers = random.sample(
+            [torch.FloatTensor([1, i, j, k]) for i, j, k in zip(
+                c_points[0], c_points[1], c_points[2])],
+            self.roi_num
+        )
```

Comment:
better use `idx = self.R.randint(low=0, high=len(c_points[0]))` to choose a index, and `roi_centers = (c_points[0][idx], c_points[1][idx], c_points[2][idx])`